raiden/network/transport/matrix/transport.py

```diff
@@ -224,6 +228,29 @@ def notify(self) -> None:
         with self._lock:
             self._notify_event.set()
 
+    def _batch_by_address_metadata(
+        self,
+    ) -> List[Tuple[List[_MessageData], Optional[AddressMetadata]]]:
+        def key_func(message_data: "_RetryQueue._MessageData") -> str:
+            address_metadata = message_data.address_metadata
+            if address_metadata is None:
+                return ""
+            uid = address_metadata.get("user_id", "")
+            return cast(str, uid)
```

Comment:
Why not simply batch by the entire `address_metadata`?
Then this could be replaced by `attrgetter("address_metadata")`

Reply:
Maybe I'm missing something, but:
`groupby( )` expects a pre-sorted list and a sort-key-function in order to work, and `address_metadata` is a dict, which can't be used for sorting unless it would be serialised first?

Reply:
Ah right. Somehow I thought it was a dataclass. 
Oh well then let's just keep it as is.